system/src/system_network_diagnostics.h

```diff
@@ -85,4 +86,36 @@ class NetworkDiagnostics {
     SimpleIntegerDiagnosticData lastError_;
 };
 
+struct NetIfDiagnostics {
+    network_interface_t interface;
+    uint32_t dnsResolutionAttempts;
+    uint32_t dnsResolutionFailures;
+    uint32_t socketConnAttempts;
+    uint32_t socketConnFailures;
+    uint32_t txBytes;
+    uint32_t rxBytes;
+    uint32_t rxTimeouts;
+    uint32_t avgPacketRoundTripTime;
+};
+
+class NetIfTester {
+public:
+    NetIfTester();
+    ~NetIfTester();
+
+    static NetIfTester* instance();
+    void testInterfaces();
+
+    const Vector<NetIfDiagnostics>* getDiagnostics();
+
+private:
+    int testInterface(NetIfDiagnostics* diagnostics);
+
+    const uint16_t UDP_ECHO_PORT = 40000;
+    const char * UDP_ECHO_SERVER_HOSTNAME = "publish-receiver-udp.particle.io";
```

Comment:
Hard coding a UDP echo server endpoint, but this should probably be a configured or stored somewhere else

Reply:
Isn't this stored in DCT?

Reply:
The device service address is stored in DCT, not this echo server that im using for testing here. 
I think long term we want to have a more official hosted/monitored service that we point to rather than this test instance.